### ISSUE TITLE
Apparently wrong correction

### ISSUE THREAD

**Issue by Average-user** [ndmitchell/hlint#454]
HLint made this suggestion to me.`

```Shell
Warning: Functor law
Found:
sort <$> (f input) `g` sort <$> x
Why not:
sort . ((f input) `g` sort) <$> x
```
But the program doesn't compile at all with the suggested version.

**Comment by ndmitchell** [comment]
Thanks for the report!

I tried to reproduce, and it worked for me. As a complete reproduction I went with:

```haskell
import Data.List
test :: (Functor f0, Ord a1, Ord a0) => (t -> t1) -> (t1 -> ([a0] -> [a0]) -> a -> [a1]) -> t -> f0 a -> f0 [a1]
test f g input x = sort <$> (f input) `g` sort <$> x
```

I got GHC to infer the type signature and write it down, to ensure I didn't give you a more restricted output. I then replaced it with the output to give:

```haskell
import Data.List
test2 :: (Functor f0, Ord a1, Ord a0) => (t -> t1) -> (t1 -> ([a0] -> [a0]) -> a -> [a1]) -> t -> f0 a -> f0 [a1]
test2 f g input x = sort . ((f input) `g` sort) <$> x
```

And it still worked.

Looking at https://github.com/exercism/haskell/pull/663/commits/7f35fcff8e598131083a18ebae91b95b014e28c5#diff-b1e1ab250f15008aa87efc403b6823c9 (which I guess is where you were starting from) I think the actual issue is that HLint gets the wrong fixity for `shouldBe`, so it gets the wrong parse tree, and then everything goes wrong. After changing HLint to learn about `shouldBe`, it then says the bracket around `(f input)` is redundant (which it is), so it will work properly in the next release.
